File: src/main/java/com/hyperboat/daily/y2024/m10/d19/LC3192.java
package com.hyperboat.daily.y2024.m10.d19;

/*
 * https://leetcode.cn/problems/minimum-operations-to-make-binary-array-elements-equal-to-one-ii/description/
 * */
public class LC3192 {

  public int minOperations(int[] nums) {
    var ans = 0;
    for (int num : nums) {
      if (((ans & 1) ^ num) == 0) {
        ans++;
      }
    }
    return ans;
  }
}

File: src/main/java/com/hyperboat/daily/y2024/m10/d19/LC3192.kt
package com.hyperboat.daily.y2024.m10.d19

internal class Solution {
    fun minOperations(nums: IntArray): Int {
        val n = nums.size
        var ans = 0
        for (i in 0 until n) {
            if (ans and 1 xor nums[i] == 0) {
                ans++
            }
        }
        return ans
    }
}